src/mlpack/bindings/python/mlpack/preprocess_json_params.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python
+"""
+preprocess_json_params.py: utility functions for json paramter preprocessing
+                           (see set_cpp_param() and get_cpp_param() methods
+                            in print_class_defn.hpp)
+
+The "process_params_out" and "process_params_in" utilities are used to handle
+interconversion between the output json from cereal and python dictionary.
+
+mlpack is free software; you may redistribute it and/or modify it under the
+terms of the 3-clause BSD license.  You should have received a copy of the
+3-clause BSD license along with mlpack.  If not, see
+http://www.opensource.org/licenses/BSD-3-Clause for more information.
+"""
+import numpy as np
+import json
+import pprint
+from copy import deepcopy
+from collections import OrderedDict
+
+def process_params_out(model, params, return_str=False):
+  '''
+  This method processes the parameters obtained from the model.
+
+  params:
+  1) model - the model to process params.
+  2) params - json parameters of the model (which we get through cereal).
+  3) return_str (bool) - if True then a pretty string version of the params is returned.
+  '''
+  # for pretty printing.
+  pp = pprint.PrettyPrinter()
+
+  # value_resolver defined later.
+  params_dic = json.loads(params, object_pairs_hook=value_resolver)
+
+  # remove "cereal_class_version".
+  cereal_class_version = [] # this stores the cereal_class_version value for all deleted pairs.
+  ref_path = []
+  # 'full_paths' will store the complete path in the dictionary for all occurrences.
+  # this will be used during the reversed process, to insert 'cereal_class_version'
+  # at the correct places to avoid any errors.
+  full_paths = []
+
+  scrub(params_dic, "cereal_class_version", cereal_class_version, full_paths, ref_path)
+
+  # storing 'cereal_class_version' occurrences paths and values.
+  model.scrubbed_params["cereal_class_version"] = {
+      "values": cereal_class_version,
+      "full_paths": full_paths,
+  }
+
+  # convert armadillo dictionary to numpy array.
+  arma_to_np(params_dic)
+
+  if return_str:
+    return params_dic, pp.pformat(params_dic)
+  else:
+    return params_dic
+
+def process_params_in(model, params_dic):
+  """
+  This function takes in a model and the parameters dictionary,
+  and returns a string that can be ingested back into the model.
+  """
+  # deepcopy to prevent changes to the user dictionary.
+  params_dic_copy = deepcopy(params_dic)
+
+  # convert numpy to armadillo.
+  np_to_arma(params_dic_copy)
+
+  # inserting scrubbed parameters back into dictionary.
+  for param_name, details in model.scrubbed_params.items():
+    for val, path in zip(details["values"], details["full_paths"]):
+      insert_in_dic(params_dic_copy, path, param_name, val)
+
+  # dumping to string. restore_value defined later.
+  params_str = json.dumps(params_dic_copy,  cls=restore_value)
```

Comment:
```suggestion
  params_str = json.dumps(params_dic_copy, cls=restore_value)
```
Tiny spacing fix. :)